src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useCallback } from 'react'
+import { Menu } from '@blueprintjs/core'
+import { FormInput, getMultiTypeFromValue, Layout, MultiTypeInputType, Text } from '@harness/uicore'
+import { get, memoize } from 'lodash-es'
+import type { GetDataError } from 'restful-react'
+
+import { ConfigureOptions } from '@common/components/ConfigureOptions/ConfigureOptions'
+import { EXPRESSION_STRING } from '@pipeline/utils/constants'
+import { useMutateAsGet } from '@common/hooks'
+
+import { Failure, useGetLastSuccessfulBuildForDocker } from 'services/cd-ng'
+import { useStrings } from 'framework/strings'
+
+import css from '../../ArtifactConnector.module.scss'
+
+const onTagInputFocus = (e: React.FocusEvent<HTMLInputElement>, fetchDigest: () => void): void => {
+  if (e?.target?.type !== 'text' || (e?.target?.type === 'text' && e?.target?.placeholder === EXPRESSION_STRING)) {
+    return
+  }
+  fetchDigest()
+}
+
+export function NoDigestResults({
+  digestError,
+
+  defaultErrorText
+}: {
+  digestError: GetDataError<Failure | Error> | null
+  defaultErrorText?: string
+}): JSX.Element {
+  const { getString } = useStrings()
+
+  const getErrorText = useCallback(() => {
+    return defaultErrorText || getString('pipeline.artifactsSelection.errors.nodigest')
+  }, [getString])
+
+  return (
+    <Text className={css.padSmall} lineClamp={1}>
+      {get(digestError, 'data.message', null) || getErrorText()}
+    </Text>
+  )
+}
+
+function ArtifactDigestField({
+  formik,
+  buildDetailsLoading,
+  expressions,
+  allowableTypes,
+  isReadonly,
+  accountIdentifier,
```

Comment:
Instead of passing`accountIdentifier, orgIdentifier,
  projectIdentifier,` values as params these can be used form url param
```suggestion
  const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
```